core/test/audits/dobetterweb/uses-http2-test.js

```diff
@@ -34,63 +63,153 @@ describe('Resources are fetched over http/2', () => {
-    expect(hosts).toEqual(new Set(['pwa.rocks']));
+    expect(hosts).toEqual(new Set(['www.example.com']));
     // make sure we flag all the rest
-    expect(result.details.items).toHaveLength(60);
+    expect(result.details.items).toHaveLength(4);
     // make sure we report savings
-    expect(result.numericValue).toMatchInlineSnapshot(`1340`);
-    expect(result.details.overallSavingsMs).toMatchInlineSnapshot(`1340`);
+    expect(result.numericValue).toMatchInlineSnapshot(`1150`);
+    expect(result.details.overallSavingsMs).toMatchInlineSnapshot(`1150`);
     // make sure we have a failing score
     expect(result.score).toBeLessThan(0.5);
+    expect(result.metricSavings).toEqual({LCP: 1150, FCP: 1150});
   });
 
   it('should ignore service worker requests', async () => {
-    const records = await NetworkRecords.compute_(artifacts.devtoolsLogs.defaultPass);
-    records.forEach(record => (record.protocol = 'HTTP/1.1'));
-    records.slice(30).forEach(record => {
-      // Force the records we're making service worker to another origin.
-      // Because it doesn't make sense to have half H2 half not to the same origin.
-      const url = record.url;
-      if (url.includes('pwa.rocks')) record.url = url.replace('pwa.rocks', 'pwa2.rocks');
-      record.fetchedViaServiceWorker = true;
-      delete record.parsedURL;
-    });
-
-    artifacts.devtoolsLogs.defaultPass = networkRecordsToDevtoolsLog(records);
+    const networkRecords = [
+      ...baseResources,
+      {
+        url: 'https://www.example.com/sw',
+        fetchedViaServiceWorker: true,
+        protocol: 'HTTP/1.1',
+        priority: 'High',
+      },
+      {
+        url: 'https://www.example.com/sw2',
+        fetchedViaServiceWorker: true,
+        protocol: 'HTTP/1.1',
+        priority: 'High',
+      },
+      {
+        url: 'https://www.example.com/sw3',
+        fetchedViaServiceWorker: true,
+        protocol: 'HTTP/1.1',
+        priority: 'High',
+      },
+      {
+        url: 'https://www.example.com/sw4',
+        fetchedViaServiceWorker: true,
+        protocol: 'HTTP/1.1',
+        priority: 'High',
+      },
+      {
+        url: 'https://www.example.com/sw6',
+        fetchedViaServiceWorker: true,
+        protocol: 'HTTP/1.1',
+        priority: 'High',
+      },
+      {
+        url: 'https://www.example.com/sw7',
+        networkRequestTime: 2000, // after FCP
+        transferSize: 500_000,
+        protocol: 'HTTP/1.1',
+        priority: 'High',
+      }];
+    artifacts.devtoolsLogs.defaultPass = networkRecordsToDevtoolsLog(networkRecords);
+
     const result = await UsesHTTP2Audit.audit(artifacts, context);
     const urls = new Set(result.details.items.map(item => item.url));
 
     // make sure we flag only the non-sw ones
-    expect(urls).not.toContain(records[30].url);
-    expect(result.details.items).toHaveLength(30);
+    expect(urls).not.toContain(new Set(['www.example.com', 'https://www.example.com/sw7']));
```

Comment:
I don't think this logic is working as you are expecting. `toContain` won't check every item in the list. It's passing because it checks if the `new Set` object itself is in the `urls` list :)

```suggestion
    expect(urls).not.toContain('https://www.example.com/');
    expect(urls).not.toContain('https://www.example.com/sw7');
```

Reply:
thanks :)